pandas/tests/plotting/frame/test_frame.py

```diff
@@ -1725,13 +1718,12 @@ def test_errorbar_plot_different_yerr_xerr_subplots(self, kind):
         )
         _check_has_errorbars(axes, xerr=1, yerr=1)
 
-    @pytest.mark.xfail(reason="Iterator is consumed", raises=ValueError)
     def test_errorbar_plot_iterator(self):
         d = {"x": np.arange(12), "y": np.arange(12, 0, -1)}
         df = DataFrame(d)
 
         # yerr is iterator
-        ax = _check_plot_works(df.plot, yerr=itertools.repeat(0.1, len(df)))
+        ax = _check_plot_works(df.plot, yerr=np.full(len(df), 0.1))
```

Comment:
Docstring for yerr says array-like; I don't think we should expect an iterator to work.

Reply:
the test name and comment a line up specifically say iterator.  i dont care about iterators either, but someone at some point did this intentionally.  may merit a whatsnew if we are officially giving up on this

Reply:
Thanks - I took another look. This goes back to when the error bars were added in #6834. The issue is actually on our end - we pass kwargs twice to two different subplots and on the 2nd the generator is exhausted. We can fix by making a copy in the test.

It seems to me this could break in the future because [the matplotlib docs](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.errorbar.html) don't say generators are supported, but we can leave as-is for now.